### PR TITLE
Refactor workflow into an action

### PR DESCRIPTION
**Pull request by ArthurBandaryk**
We call the `composite` action (`action.yml`) from `3rdparty/dev-tools` submodule. This action helps us to check files for correct code style, no matter if they have `.cc | .h ` or `.bzl` extensions.

### CODE REVIEW

File: .github/workflows/check_code_style.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Initialize and update submodule
+        run: git submodule update --init
+
+      # Call the composite action to check files
+      # for correct code style. This action (action.yml)
+      # is in `3rdparty/dev-tools` submodule.
+      - uses: 3rdparty/dev-tools@main
```
Comment (benh) [inline .github/workflows/check_code_style.yml:23]: This is unfortunate naming. It looks like though that we _can_ have multiple actions per repository, see [here](https://github.community/t/possible-to-have-multiple-actions-in-a-single-repo/159823). Then we can put `action.yml` into the `check-code-style` directory of `dev-tools` and do:
```suggestion
      - uses: 3rdparty/dev-tools/check-code-style@main
```

File: .github/workflows/check_code_style.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      - "main" 
+
+jobs:
+  check_code_style:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Initialize and update submodule
+        run: git submodule update --init
```
Comment (benh) [inline .github/workflows/check_code_style.yml:18]: Should we also be recursive?
```suggestion
        run: git submodule update --init --recursive
```

**Review by benh — CHANGES_REQUESTED** [review]

File: .github/workflows/check_code_style.yml (hunk 2 above)
Comment (ArthurBandaryk) [reply .github/workflows/check_code_style.yml:18]: Yep, I think this is a great idea!

File: .github/workflows/check_code_style.yml (hunk 1 above)
Comment (ArthurBandaryk) [reply .github/workflows/check_code_style.yml:23]: This [PR](https://github.com/3rdparty/dev-tools/pull/14) is waiting for you @benh !

Comment (benh) [reply .github/workflows/check_code_style.yml:23]: Ok approved! Now you can update and commit!

File: .github/workflows/check_code_style.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+      # Call the composite action to check files
+      # for correct code style. This action (action.yml)
```
Comment (benh) [inline .github/workflows/check_code_style.yml:21]:
```suggestion
      # for correct code style. This action
```

**Review by benh — APPROVED** [review]

**Pull request merged**